File: DailyLeetcode/19Aug/NumberofZeroFilledSubarrays.java
// Solved using a single pass approach
//Time Complexity: O(n)
//Space Complexity: O(1)
class Solution {
    public long zeroFilledSubarray(int[] nums) {
        long count=0, ans=0;
        for(int num: nums){
            ans= (num ==0)? ans +1 :0;
            count+=ans;
        }
        return count;
    }
}

File: DailyLeetcode/19Aug/NumberofZeroFilledSubarrays.cpp
//Time Complexity=O(n)
//Space Complexity=O(1)
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        long long cnt=0;
        long long streak=0;

        for(auto & num:nums){
            if(num==0){
                streak++;
                cnt+=streak;
            }
            else{
                streak=0;
            }
        }
        return cnt;
    }
};